### PR TITLE
Add COMMENT_TITLE env var for issue comment title

### PR DESCRIPTION
**Pull request by mas-wtag**
This PR allows hard coded issue comment title `Unit Test Results` to be changed when `COMMENT_TITLE` env var is defined.

### CODE REVIEW

**Comment by EnricoMi** [comment]
Thanks for this PR. I was wondering if that comment title should be configurable independently from the check name, maybe through a `COMMENT_TITLE` env var. The check run name might make sense in the Actions tab but less in a PR comment and vice versa. What do you think?

**Comment by mas-wtag** [comment]
Hi @EnricoMi , thanks for taking a look to the PR.
> I was wondering if that comment title should be configurable independently from the check name, maybe through a COMMENT_TITLE env var.
- Sure, I also think that would be a good way forward. I'll make the necessary adjustments in code.

**Comment by EnricoMi** [comment]
Issue #42 made me think the comment title should use `CHECK_NAME` if no `COMMENT_TITLE` is given, otherwise `COMMENT_TITLE`. This allows to use different title and check run name if needed, or the same if sufficient.

**Comment by mas-wtag** [comment]
> Issue #42 made me think the comment title should use `CHECK_NAME` if no `COMMENT_TITLE` is given, otherwise `COMMENT_TITLE`. This allows to use different title and check run name if needed, or the same if sufficient.

@EnricoMi please have a look and let me know if it's all good 👍

File: publish_unit_test_results.py [hunk 1]
```diff
@@ -735,7 +735,7 @@ def write_stats_file(stats, filename) -> None:
         f.write(json.dumps(stats))
 
 
-def main(token: str, event: dict, repo: str, commit: str, files_glob: str, check_name: str,
+def main(token: str, event: dict, repo: str, commit: str, files_glob: str, issue_comment_title: str, check_name: str,
```
Comment (EnricoMi) [inline publish_unit_test_results.py:738]: Can we call this `comment_title` throughout the file? The `title` inside `publish_comment` is fine. And can `check_name` come first, then `comment_title`?

**Review by EnricoMi — CHANGES_REQUESTED** [review]
Only minor changes, looks good.

The README.md also needs to mention the new option. Please add it to the example in https://github.com/EnricoMi/publish-unit-test-result-action/blob/master/README.md#using-this-action and put a short description in a new paragraph below `check_name` and mention that it defaults to `check_name`s value.

File: publish_unit_test_results.py [hunk 2]
```diff
@@ -693,7 +693,7 @@ def hide_comment(comment_node_id) -> bool:
         comments = list([comment for comment in comments
                          if comment.get('author', {}).get('login') == 'github-actions'
                          and comment.get('isMinimized') is False
-                         and comment.get('body', '').startswith('## Unit Test Results\n')
+                         and comment.get('body', '').startswith('## {}\n'.format(check_name))
```
[the remark this answers is not in the document]
Comment (mas-wtag) [reply publish_unit_test_results.py]: good catch :)

File: publish_unit_test_results.py (hunk 1 above)
Comment (mas-wtag) [reply publish_unit_test_results.py]: Sure, i just thought to name it differently since it shadows the name from the outer scope

Comment (EnricoMi) [reply publish_unit_test_results.py]: You are right. I am going to refactor that code soon, no shadowing then.

**Comment by EnricoMi** [comment]
All looks good, but one unit test fails, not in your PR #33, but when I pulled your forked branch into my repo for testing, the same commit suddenly fails (#45). I will care about that later, but please fix `get_short_summary` not knowing `check_name` when called form a unit test (because it is defined in `__main__`). The `default` argument should be provided to `get_short_summary` as an argument.

**Comment by EnricoMi** [comment]
The action works as expected:

- only check name set: https://github.com/EnricoMi/publish-unit-test-result-action/pull/44#issuecomment-717142427
- comment title set: https://github.com/EnricoMi/publish-unit-test-result-action/pull/44#issuecomment-717143393

**Review by EnricoMi — APPROVED** [review]

**Comment by EnricoMi** [comment]
Thanks for the contribution!

**Pull request merged**